internal/providers/pluginfw/resources/cluster/data_cluster.go

```diff
@@ -0,0 +1,128 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/compute"
+	"github.com/databricks/terraform-provider-databricks/common"
+	pluginfwcommon "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/common"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/converters"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/tfschema"
+	"github.com/databricks/terraform-provider-databricks/internal/service/compute_tf"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func DataSourceCluster() datasource.DataSource {
+	return &ClusterDataSource{}
+}
+
+var _ datasource.DataSourceWithConfigure = &ClusterDataSource{}
+
+type ClusterDataSource struct {
+	Client *common.DatabricksClient
+}
+
+type ClusterInfo struct {
+	ClusterId   types.String               `tfsdk:"cluster_id" tf:"optional,computed"`
+	Name        types.String               `tfsdk:"cluster_name" tf:"optional,computed"`
+	ClusterInfo *compute_tf.ClusterDetails `tfsdk:"cluster_info" tf:"optional,computed"`
+}
+
+func (d *ClusterDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = "databricks_cluster_pluginframework"
+}
+
+func (d *ClusterDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: tfschema.DataSourceStructToSchemaMap(ClusterInfo{}, nil),
+	}
+}
+
+func (d *ClusterDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if d.Client == nil {
+		d.Client = pluginfwcommon.ConfigureDataSource(req, resp)
+	}
+}
+
+func ValidateClustersList(ctx context.Context, clusters []compute_tf.ClusterDetails, clusterName string) diag.Diagnostics {
```

Comment:
Why does this need to be exported?

Reply:
Regarding having different directory for unit test — I was thinking that there would be minimum 9 files at top level (2 data sources, 1 resource and their unit and integration tests) so navigating might become difficult. But naming them _acc_test.go  to differentiate also works.

We can export ValidateClustersList (Unit tests) by keeping it in same <package> but acceptance tests have to be in <package>_test due to cyclic dependency.